The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/9474205817/job/26103337127) [(plain)](https://github.com/rust-lang/rust/commit/f58c65a7b815b5fe0803a56d1c0334597e4d06d7/checks/26103337127/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:a5ac7e51b41094c92402da3b24376905380afc29)
Download action repository 'actions/upload-artifact@v4' (SHA:65462800fd760344b1a7b4382951275a0abb4808)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.413   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.429 Collecting boolean-py==4.0
#13 0.432   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.443 Collecting chardet==5.1.0
---
#13 3.477 Building wheels for collected packages: reuse
#13 3.478   Building wheel for reuse (pyproject.toml): started
#13 3.806   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 3.807   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 3.807   Stored in directory: /tmp/pip-ephem-wheel-cache-ar5h9xo7/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 3.810 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 3.832   Attempting uninstall: setuptools
#13 3.833     Found existing installation: setuptools 59.6.0
#13 3.834     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
---
#13 5.083   Downloading virtualenv-20.26.2-py3-none-any.whl (3.9 MB)
#13 5.155      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.9/3.9 MB 56.1 MB/s eta 0:00:00
#13 5.205 Collecting filelock<4,>=3.12.2
#13 5.209   Downloading filelock-3.14.0-py3-none-any.whl (12 kB)
#13 5.239 Collecting platformdirs<5,>=3.9.1
#13 5.242   Downloading platformdirs-4.2.2-py3-none-any.whl (18 kB)
#13 5.259 Collecting distlib<1,>=0.3.7
#13 5.263   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.269      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 100.5 MB/s eta 0:00:00
#13 5.357 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 5.521 Successfully installed distlib-0.3.8 filelock-3.14.0 platformdirs-4.2.2 virtualenv-20.26.2
#13 DONE 5.6s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      196544 kB
DirectMap2M:     7143424 kB
DirectMap1G:    11534336 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.04s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/d0227c6a19c2d6e8dceb87c7a2776dc2b10d2a04/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-d0227c6a19c2d6e8dceb87c7a2776dc2b10d2a04-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 26.01s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/core/src/ptr/mod.rs at line 728:
 #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
 #[allow(fuzzy_provenance_casts)] // this *is* the explicit provenance API one should use instead
 pub fn with_exposed_provenance<T>(addr: usize) -> *const T
-        T: Sized,
+where
+    T: Sized,
 {
 {
     // FIXME(strict_provenance_magic): I am magic and should be a compiler intrinsic.
     addr as *const T
##[error]Diff in /checkout/library/core/src/ptr/mod.rs at line 768:
 #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
 #[allow(fuzzy_provenance_casts)] // this *is* the explicit provenance API one should use instead
 pub fn with_exposed_provenance_mut<T>(addr: usize) -> *mut T
-        T: Sized,
+where
+    T: Sized,
 {
 {
     // FIXME(strict_provenance_magic): I am magic and should be a compiler intrinsic.
     addr as *mut T
##[error]Diff in /checkout/library/core/src/ptr/mod.rs at line 1067:
     // because we don't want to pessimize things like slices of SIMD vectors.)
     if mem::align_of::<T>() <= mem::size_of::<usize>()
         && (!mem::size_of::<T>().is_power_of_two()
-        || mem::size_of::<T>() > mem::size_of::<usize>() * 2)
+            || mem::size_of::<T>() > mem::size_of::<usize>() * 2)
     {
         attempt_swap_as_chunks!(usize);
         attempt_swap_as_chunks!(u8);
##[error]Diff in /checkout/library/core/src/ptr/mod.rs at line 1405:
 #[stable(feature = "ptr_unaligned", since = "1.17.0")]
 #[rustc_const_stable(feature = "const_ptr_read", since = "1.71.0")]
 #[rustc_allow_const_fn_unstable(
-const_maybe_uninit_as_mut_ptr,
-const_intrinsic_copy
+    const_mut_refs,
+    const_maybe_uninit_as_mut_ptr,
+    const_maybe_uninit_as_mut_ptr,
+    const_intrinsic_copy
 )]
 #[cfg_attr(miri, track_caller)] // even without panics, this helps for Miri backtraces
 #[rustc_diagnostic_item = "ptr_read_unaligned"]
##[error]Diff in /checkout/library/core/src/ptr/mod.rs at line 2205:
 #[rustc_macro_transparency = "semitransparent"]
 #[allow_internal_unstable(raw_ref_op)]
 pub macro addr_of($place:expr) {
-&raw const $place
+    &raw const $place
 
 
 /// Create a `mut` raw pointer to a place, without creating an intermediate reference.
##[error]Diff in /checkout/library/core/src/ptr/mod.rs at line 2293:
 #[rustc_macro_transparency = "semitransparent"]
 #[allow_internal_unstable(raw_ref_op)]
 pub macro addr_of_mut($place:expr) {
-&raw mut $place
+    &raw mut $place
 
 
fmt error: Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/compiler/rustc_codegen_cranelift/src/abi/comments.rs" "/checkout/compiler/rustc_codegen_cranelift/src/abi/returning.rs" "/checkout/compiler/rustc_codegen_cranelift/src/abi/pass_mode.rs" "/checkout/compiler/rustc_codegen_cranelift/src/cast.rs" "/checkout/compiler/rustc_codegen_cranelift/src/lib.rs" "/checkout/compiler/rustc_codegen_cranelift/src/trap.rs" "/checkout/compiler/rustc_codegen_cranelift/src/main_shim.rs" "/checkout/compiler/rustc_codegen_cranelift/src/toolchain.rs" "/checkout/compiler/rustc_codegen_cranelift/src/num.rs" "/checkout/compiler/rustc_codegen_cranelift/src/global_asm.rs" "/checkout/compiler/rustc_codegen_cranelift/src/inline_asm.rs" "/checkout/compiler/rustc_codegen_cranelift/src/constant.rs" "/checkout/library/core/src/ptr/metadata.rs" "/checkout/library/core/src/ptr/mod.rs" "/checkout/library/core/src/ptr/unique.rs" "/checkout/library/core/src/ptr/const_ptr.rs" "/checkout/library/core/src/ptr/alignment.rs" "/checkout/library/core/src/ptr/non_null.rs" "/checkout/library/core/src/ptr/mut_ptr.rs" "/checkout/library/core/src/pat.rs" "/checkout/library/core/src/clone.rs" "/checkout/library/core/src/unicode/mod.rs" "/checkout/library/core/src/unicode/unicode_data.rs" "/checkout/library/core/src/unicode/printable.rs" "/checkout/compiler/rustc_codegen_cranelift/src/debuginfo/mod.rs" "/checkout/compiler/rustc_codegen_cranelift/src/debuginfo/line_info.rs" "/checkout/compiler/rustc_codegen_cranelift/src/debuginfo/emit.rs" "/checkout/compiler/rustc_codegen_cranelift/src/debuginfo/unwind.rs" "/checkout/compiler/rustc_codegen_cranelift/src/debuginfo/types.rs" "/checkout/compiler/rustc_codegen_cranelift/src/debuginfo/object.rs" "/checkout/compiler/rustc_codegen_cranelift/src/compiler_builtins.rs" "/checkout/compiler/rustc_codegen_cranelift/src/codegen_i128.rs" "/checkout/compiler/rustc_codegen_cranelift/src/vtable.rs" "/checkout/compiler/rustc_codegen_cranelift/src/analyze.rs" "/checkout/compiler/rustc_codegen_cranelift/src/pointer.rs" "/checkout/compiler/rustc_codegen_cranelift/src/common.rs" "/checkout/compiler/rustc_codegen_cranelift/src/discriminant.rs" "/checkout/compiler/rustc_codegen_cranelift/src/concurrency_limiter.rs" "/checkout/compiler/rustc_codegen_cranelift/src/driver/mod.rs" "/checkout/compiler/rustc_codegen_cranelift/src/driver/jit.rs" "/checkout/compiler/rustc_codegen_cranelift/src/driver/aot.rs" "/checkout/compiler/rustc_codegen_cranelift/src/archive.rs" "/checkout/compiler/rustc_codegen_cranelift/src/unsize.rs" "/checkout/compiler/rustc_codegen_cranelift/src/base.rs" "/checkout/compiler/rustc_codegen_cranelift/src/value_and_place.rs" "/checkout/compiler/rustc_codegen_cranelift/src/intrinsics/mod.rs" "/checkout/compiler/rustc_codegen_cranelift/src/intrinsics/llvm_x86.rs" "/checkout/compiler/rustc_codegen_cranelift/src/intrinsics/llvm.rs" "/checkout/compiler/rustc_codegen_cranelift/src/intrinsics/simd.rs" "/checkout/compiler/rustc_codegen_cranelift/src/intrinsics/llvm_aarch64.rs" "/checkout/compiler/rustc_codegen_cranelift/src/config.rs" "/checkout/compiler/rustc_codegen_cranelift/src/allocator.rs" "/checkout/library/core/src/num/fmt.rs" "/checkout/library/core/src/num/int_macros.rs" "/checkout/library/core/src/num/int_log10.rs" "/checkout/library/core/src/num/mod.rs" "/checkout/library/core/src/num/saturating.rs" "/checkout/library/core/src/num/overflow_panic.rs" "/checkout/library/core/src/num/f64.rs" "/checkout/library/core/src/num/diy_float.rs" "/checkout/library/core/src/num/uint_macros.rs" "/checkout/compiler/rustc_codegen_cranelift/src/optimize/mod.rs" "/checkout/library/core/src/num/bignum.rs" "/checkout/compiler/rustc_codegen_cranelift/src/abi/mod.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Jun 11 23:40:48 UTC 2024
  network time: Tue, 11 Jun 2024 23:40:48 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/core/src/ptr/mod.rs
+++ b/library/core/src/ptr/mod.rs
@@ -237,7 +237,7 @@
 //! pointer. For code which *does* cast a usize to a pointer, the scope of the change depends
 //! on exactly what you're doing.
 //!
-//! In general you just need to make sure that if you want to convert a usize address to a
+//! In general, you just need to make sure that if you want to convert a usize address to a
 //! pointer and then use that pointer to read/write memory, you need to keep around a pointer
 //! that has sufficient provenance to perform that read/write itself. In this way all of your
 //! casts from an address to a pointer are essentially just applying offsets/indexing.
@@ -309,7 +309,7 @@
 //!   i.e. the usual "ZSTs are fake, do what you want" rules apply *but* this only applies
 //!   for actual forgery (integers cast to pointers). If you borrow some struct's field
 //!   that *happens* to be zero-sized, the resulting pointer will have provenance tied to
-//!   that allocation and it will still get invalidated if the allocation gets deallocated.
+//!   that allocation, and it will still get invalidated if the allocation gets deallocated.
 //!   In the future we may introduce an API to make such a forged allocation explicit.
 //!
 //! * [`wrapping_offset`][] a pointer outside its provenance. This includes pointers
@@ -698,7 +698,7 @@ pub const fn dangling_mut<T>() -> *mut T {
 ///
 /// If there is no 'exposed' provenance that justifies the way this pointer will be used,
 /// the program has undefined behavior. In particular, the aliasing rules still apply: pointers
-/// and references that have been invalidated due to aliasing accesses cannot be used any more,
+/// and references that have been invalidated due to aliasing accesses cannot be used anymore,
 /// even if they have been exposed!
 ///
 /// Note that there is no algorithm that decides which provenance will be used. You can think of this
@@ -1097,7 +1097,7 @@ const unsafe fn swap_nonoverlapping_simple_untyped<T>(x: *mut T, y: *mut T, coun
         // If we end up here, it's because we're using a simple type -- like
         // a small power-of-two-sized thing -- or a special type with particularly
         // large alignment, particularly SIMD types.
-        // Thus we're fine just reading-and-writing it, as either it's small
+        // Thus, we're fine just reading-and-writing it, as either it's small
         // and that works well anyway or it's special and the type's author
         // presumably wanted things to be done in the larger chunk.
 
@@ -1290,7 +1290,7 @@ pub const unsafe fn read<T>(src: *const T) -> T {
     // provides enough information to know that this is a typed operation.
 
     // However, as of March 2023 the compiler was not capable of taking advantage
-    // of that information.  Thus the implementation here switched to an intrinsic,
+    // of that information. Thus, the implementation here switched to an intrinsic,
     // which lowers to `_0 = *src` in MIR, to address a few issues:
     //
     // - Using `MaybeUninit::assume_init` after a `copy_nonoverlapping` was not
@@ -1570,7 +1570,7 @@ pub const unsafe fn write<T>(dst: *mut T, src: T) {
 /// As a result, using `&packed.unaligned as *const FieldType` causes immediate
 /// *undefined behavior* in your program.
 ///
-/// Instead you must use the [`ptr::addr_of_mut!`](addr_of_mut)
+/// Instead, you must use the [`ptr::addr_of_mut!`](addr_of_mut)
 /// macro to create the pointer. You may use that returned pointer together with
 /// this function.
 ///

